chrome/content/zotero/elements/annotationRow.js

```diff
@@ -96,7 +96,7 @@
 			if (this._annotation.annotationText) {
 				let text = document.createElement('div');
 				text.classList.add('quote');
-				text.textContent = this._annotation.annotationText;
+				text.innerHTML = this._annotation.annotationText;
```

Comment:
That's fair, I didn't think about it. I stumbled upon this [mozilla util](https://searchfox.org/mozilla-central/source/parser/html/nsIParserUtils.idl#100) that might be helpful here. From what I can tell, it strips attributes and non-style tags from components within a string.

So if we do:

```
let parserUtils = Cc["@mozilla.org/parserutils;1"].getService(Ci.nsIParserUtils);
parserUtils.sanitize("<div attribute='something'>this</div> <unknown>is</unknown> a <b>test</b><script>console.log('!')</script> <img src='test'>", Ci.nsIParserUtils.SanitizerDropForms | Ci.nsIParserUtils.SanitizerDropMedia)
```

We'll get `<html><head></head><body><div>this</div> is  a <b>test</b> </body></html>`. I pushed a commit with that. What do you think? Or, if it's better, I can revert it back to textcontent for now and we can tackle it separately.

Reply:
It looks like itemTree already does essentially [the same thing as the reader](https://github.com/zotero/zotero/blob/f8970b92de368a0ddb8887a1ba7dfa3c0548bdd4/chrome/content/zotero/itemTree.jsx#L2752) to properly render the title column. Now it feels even less nice to duplicate reader's logic... Would it make sense to pull out this code from `itemTree` into, maybe, `Zotero.Utilities.Internal` so that `itemTree` could share it with `annotation-row`?

Reply:
Ah, yes, I see it. Then the current rendering of annotation titles as a primary cell of the table is not right as well. Then I suppose the best way is indeed to just copy over the logic from the reader (into Zotero.Utilities.Internal as well?) and use it for annotation rows of the itemTree and the `annotation-row` component.

Reply:
That makes a lot of sense! Trying to do it now is almost a guarantee to create conflicts. I pushed a tweak to strip all html tags for now and then, once these PRs are merged, I can move over the rendering logic from the reader into that new utilities function as another mode.